app/models/invenio_rdm_record_converter.rb

```diff
@@ -79,33 +79,33 @@ def extra_data
     if !@generic_file.based_near.empty?
       data["presentation_location"] = @generic_file.based_near
     end
-    data["permissions"] = file_permissions
 
+    data["permissions"] = file_permissions
     data
   end
 
   def owner_info(depositor)
     user = User.find_by(username: depositor)
 
     if user
-      {user.username => user_email(user)}
+      {user.username => user_email(user.email)}
     else
       {"unknown": "unknown"}
     end
   end
 
   def file_permissions
-    permission_data = Hash.new
-
+    permission_data = {}
     permission_data["owner"] = owner_info(@generic_file.depositor)
 
     @generic_file.permissions.each do |permission|
-      permission_data[permission.access] ||= Hash.new
+      permission_data[permission.access] ||= {}
 
-      if @role_store[permission.agent_name]
-        permission_data[permission.access].merge!(@role_store[permission.agent_name])
+      if role_user_data = @role_store[permission.agent_name]
+        role_user_data.update(role_user_data) { |user_id, email| user_email(email) }
```

Comment:
So I can follow: what is the shape of `@role_store`? And would it be simpler at this point to modify `https://github.com/galterlibrary/digital-repository/blob/57c85146b6853f71f39490363f6974e5ca77fc31/app/models/role_store.rb#L8` to have the joshelder change in there directly at this point?

Reply:
> what is the shape of `@role_store`?
Not sure exactly what you mean, but it's a hash with a role's name as the key and a value that is itself a hash with a user's NWU NetId as the key and a user's email as the value.

In code:
```
{
  role_name: {
    net_id: "user_email",
    another_net_id: "another_user_email"
  },
  another_role_name {
    yet_net_id:  "yet_another_user_email"
  }
}
```

I thought about making the change at the line you're pointing to, but I figured since we already have the function in the  converter it would be easiest to just reuse it there. I don't have strong opinions so if you would rather I move it to the `role_store` model that works for me.

Reply:
Thanks. It's not a big inclination, but I think having `normalize_email` in some `utils.rb` and reusing it in this file and in https://github.com/galterlibrary/digital-repository/blob/57c85146b6853f71f39490363f6974e5ca77fc31/app/models/role_store.rb#L8 file would be good. 

The current update pattern is strange to me: why would we update in-place `@role_store` if we always update it in-place? (and we could do that just once). With the troubles we've been having with this, I would rather we sidestep quirks if we can. I'll approve in advance though.

Reply:
Not to stir the pot, but after role_store creation, I would have updated josh elder's email before the export.
https://github.com/galterlibrary/digital-repository/blob/master/lib/scripts/repo_export_in_batches.rb#L30